tooling/image-updater/internal/output/output.go

```diff
@@ -0,0 +1,165 @@
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package output
+
+import (
+	"fmt"
+	"io"
+	"strings"
+	"text/template"
+
+	"github.com/Azure/ARO-HCP/tooling/image-updater/internal/yaml"
+)
+
+// summaryTemplate is a Go template for the summary output.
+// Available fields:
+//   - .TotalImages: total number of images processed
+//   - .UpdatedCount: number of images updated
+//   - .DryRun: whether this is a dry-run
+const summaryTemplate = `
+Summary
+-------
+{{- if .DryRun }}
+Total images checked: {{ .TotalImages }}
+Updates available:    {{ .UpdatedCount }}
+
+This was a dry-run. No files were modified.
+{{- else }}
+Total images checked: {{ .TotalImages }}
+Images updated:       {{ .UpdatedCount }}
+{{- end }}
+`
+
+// commitMessageTemplate is a Go template for the commit message.
+// Available fields in .Updates array:
+//   - .Name: image name
+//   - .OldSHA: old digest (truncated)
+//   - .NewSHA: new digest (truncated)
+//   - .Version: version tag
+//   - .Timestamp: build timestamp
+const commitMessageTemplate = `Updated images in target config files:
+
+| Image | Old SHA | New SHA | Version | Timestamp |
+|-------|---------|---------|---------|-----------|
+{{- range .Updates }}
+| {{ .Name }} | {{ .OldSHA }} | {{ .NewSHA }} | {{ .Version }} | {{ .Timestamp }} |
+{{- end }}
+`
+
+// summaryData holds data for the summary template
+type summaryData struct {
+	TotalImages  int
+	UpdatedCount int
+	DryRun       bool
+}
+
+// updateData holds data for a single update in the commit message
+type updateData struct {
+	Name      string
+	OldSHA    string
+	NewSHA    string
+	Version   string
+	Timestamp string
+}
+
+// commitMessageData holds data for the commit message template
+type commitMessageData struct {
+	Updates []updateData
+}
+
+// PrintSummary prints a formatted summary of the update operation
+func PrintSummary(w io.Writer, totalImages, updatedCount int, dryRun bool) {
+	tmpl, err := template.New("summary").Parse(summaryTemplate)
+	if err != nil {
+		fmt.Fprintf(w, "error parsing summary template: %v\n", err)
+		return
+	}
+
+	data := summaryData{
+		TotalImages:  totalImages,
+		UpdatedCount: updatedCount,
+		DryRun:       dryRun,
+	}
+
+	if err := tmpl.Execute(w, data); err != nil {
+		fmt.Fprintf(w, "error executing summary template: %v\n", err)
+	}
+}
+
+// GenerateCommitMessage creates a markdown table commit message for the updated images
+func GenerateCommitMessage(updates map[string][]yaml.Update) string {
+	var allUpdates []yaml.Update
+	for _, updates := range updates {
+		allUpdates = append(allUpdates, updates...)
+	}
+
+	if len(allUpdates) == 0 {
+		return ""
+	}
+
+	// Deduplicate updates by image name
+	seen := make(map[string]bool)
```

Comment:
please use the sets library from k8s, we already import it

Reply:
Spring maps replaced by Sets in https://github.com/Azure/ARO-HCP/pull/3504/commits/c9d0b4b582f97d38f3ce0103217340caadc2ddee